app/src/main/java/org/mozilla/fenix/home/pocket/PocketCategoriesViewHolder.kt

```diff
@@ -49,11 +49,17 @@ class PocketCategoriesViewHolder(
             composeView.resources.getDimensionPixelSize(R.dimen.home_item_horizontal_margin)
         composeView.setPadding(horizontalPadding, 0, horizontalPadding, 0)
 
+        val homeScreenReady = components.appStore
+            .observeAsComposableState { state -> state.firstFrameDrawn }.value ?: false
+
         val categories = components.appStore
             .observeAsComposableState { state -> state.pocketStoriesCategories }.value
         val categoriesSelections = components.appStore
             .observeAsComposableState { state -> state.pocketStoriesCategoriesSelections }.value
 
+        // Perceived performance can be improved by waiting to draw pocket content until after the
+        // first frame is drawn. See https://github.com/mozilla-mobile/fenix/issues/25545 for details.
```

Comment:
non-blocking: I'd prefer if this comment elaborated on why this issue happens and how this solution addresses the issue because:
1. devs won't necessarily go to the issue even if the information would be useful to them
2. if the solution changes they're not likely to update the issue but they'd update the explanation in a comment
3. there is generally a lot of irrelevant context in issues that make digging for the answers to those two questions challenging.

More generally, I prefer making the source code an elaborate source of truth and leaving the issue tracker as a supplement because the source tends to be more accurate & to-the-point.